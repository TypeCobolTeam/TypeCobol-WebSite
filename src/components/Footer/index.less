.tc-footer {color:@text-color-dark}