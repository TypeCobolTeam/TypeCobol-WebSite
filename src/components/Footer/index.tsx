import * as React from "react"

// @ts-ignore
// import logoEI from "@content/images/EI.png";

import { Layout } from "antd"
const { Footer } = Layout

class CFooter extends React.Component<{}, {}> {
  public render() {
    return <Footer style={{ textAlign: "center" }}>TypeCobol 2019</Footer>
  }
}

export default CFooter
