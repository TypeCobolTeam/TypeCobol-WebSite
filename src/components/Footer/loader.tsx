import * as React from "react"

interface FooterLoaderProps {
  lang: string
}

interface FooterLoaderStates {
  module: React.FunctionComponent
}

class FooterLoader extends React.Component<
  FooterLoaderProps,
  FooterLoaderStates
> {
  constructor(props) {
    super(props)
    this.state = {
      module: null,
    }
  }

  async componentDidMount() {
    const { lang } = this.props
    const { default: module } = await import(
      `../../../content/i18n/${lang}/footer`
    )
    this.setState({ module })
  }

  render() {
    const { module } = this.state
    if (module) return React.createElement(module)
    return <p>Loading...</p>
  }
}

export default FooterLoader
