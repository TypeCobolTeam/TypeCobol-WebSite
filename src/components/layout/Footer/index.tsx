import * as React from "react";

import "./index.scss";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

class Footer extends React.Component<any, any> {
  constructor(props: any) {
    super(props);
  }

  public render() {
    return (
      <div>
        <p>
          Made with
          <FontAwesomeIcon icon="heart" style={{ margin: "5px" }} />
          by <a>TypeCobol Team</a>
        </p>
        <p>
          The TypeCobol project is licensed under{" "}
          <a target="_blank" href="https://spdx.org/licenses/CECILL-C.html">
            CeCILL-C
          </a>
          .
        </p>
        <p>
          The website content is licensed under{" "}
          <a target="_blank" href="https://spdx.org/licenses/CECILL-C.html">
            CeCILL-C
          </a>
          .
        </p>
      </div>
    );
  }
}

export default Footer;
