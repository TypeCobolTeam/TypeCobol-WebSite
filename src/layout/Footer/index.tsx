import * as React from "react";

import "./index.scss";
import { Container } from "bloomer/lib/layout/Container";
import { Footer as BFooter } from "bloomer/lib/layout/Footer";
import { Content } from "bloomer/lib/elements/Content";
import { Columns } from "bloomer/lib/grid/Columns";
import { Column } from "bloomer/lib/grid/Column";
import { Icon } from "bloomer/lib/elements/Icon";

class Footer extends React.Component<any, any> {
  constructor(props: any) {
    super(props);
  }

  public render() {
    return (
      <BFooter id="footer">
        <Container>
          <Content>
            <Columns>
              <Column>
                <p>
                  Made with
                  <Icon className="fa fa-heart" style={{ margin: "5px" }} />
                  by <a>TypeCobol Team</a>
                </p>
              </Column>
            </Columns>
            <Content isSize="small">
              <p>
                The TypeCobol project is licensed under{" "}
                <a
                  target="_blank"
                  href="https://spdx.org/licenses/CECILL-C.html"
                >
                  CeCILL-C
                </a>
                .
              </p>
              <p>
                The website content is licensed under{" "}
                <a
                  target="_blank"
                  href="https://spdx.org/licenses/CECILL-C.html"
                >
                  CeCILL-C
                </a>
                .
              </p>
            </Content>
          </Content>
        </Container>
      </BFooter>
    );
  }
}

export default Footer;
