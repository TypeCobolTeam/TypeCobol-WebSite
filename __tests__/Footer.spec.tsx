import * as React from "react"

// @ts-ignore
import Footer from "@components/Footer"
import { render } from "enzyme"

describe("Footer", () => {
  it("renders correctly", () => {
    const wrapper = render(<Footer />)
    expect(wrapper).toMatchSnapshot()
  })
  it(`should contain site name`, () => {
    const wrapper = render(<Footer />)
    expect(wrapper.text()).toMatch(/typecobol/i)
  })
  it(`should contain a copyright notice`, () => {
    const wrapper = render(<Footer />)
    expect(wrapper.text()).toMatch(/copyright/i)
  })
  it(`should have a link to homepage (/)`, () => {
    const wrapper = render(<Footer />)
    expect(wrapper.find("a[href='/']")).toBeTruthy()
  })
})
