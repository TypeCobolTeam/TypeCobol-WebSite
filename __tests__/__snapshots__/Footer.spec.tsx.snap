// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "tc-footer ant-layout-footer",
      "style": "text-align:center",
    },
    "children": Array [
      Object {
        "data": "TypeCobol 2019",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "footer",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
